### DISCUSSION TITLE
[Feature] Kodi Viewer App

### DISCUSSION THREAD

**Discussion by lmgithu** [immich-app/immich#9388] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi, is there any way currently available to browse my photos stored on Immich from Kodi? The best I managed to do so far is to share the library on a file level, install HEIC plugin to Kodi and hope for the best, but it is extremely slow and I can't see albums, etc. I was wondering whether there is any official or community-driven solution available? I think it would be nice to have it. Thank you!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Lipown** [comment]
Would be lovely. I would love ability when users will create album to see it there.
Prior Immich users just created folders on Samba share where Kodi has the access.

**Reply by wunderbaum** [reply]
What about a _virtual_ samba-server where immich is creating directories for albums and so on?

**Comment by bpawnzZ** [comment]
I second this..  this would be amazing

**Comment by vladd11** [comment]
If somebody want to try, you can use my plugin
https://github.com/vladd11/immich-kodi/releases
It have very basic features and may not be production-ready. And you should download & install it manually.

**Reply by bpawnzZ** [reply]
good deal.. going to check it out

**Comment by kevin201** [comment]
This screensaver works well too: https://forum.kodi.tv/showthread.php?tid=380031
